modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -1650,6 +1656,9 @@ void GDScriptAnalyzer::resolve_function_signature(GDScriptParser::FunctionNode *
 			return_type.type_source = GDScriptParser::DataType::INFERRED;
 			return_type.kind = GDScriptParser::DataType::VARIANT;
 			p_function->set_datatype(return_type);
+#ifdef DEBUG_ENABLED
+			parser->push_warning(p_function, GDScriptWarning::UNTYPED_DECLARATION, "Function", function_visible_name);
+#endif
```

Comment:
The warning is not triggered for `_init()` and `_static_init()` (you can only specify `void` and no other type, but I think the warning still makes sense). Please move this after the `else`.

<details>
<summary>diff</summary>

```diff
@@ -1656,9 +1656,6 @@ void GDScriptAnalyzer::resolve_function_signature(GDScriptParser::FunctionNode *
 			return_type.type_source = GDScriptParser::DataType::INFERRED;
 			return_type.kind = GDScriptParser::DataType::VARIANT;
 			p_function->set_datatype(return_type);
-#ifdef DEBUG_ENABLED
-			parser->push_warning(p_function, GDScriptWarning::UNTYPED_DECLARATION, "Function", function_visible_name);
-#endif
 		}
 
 #ifdef TOOLS_ENABLED
@@ -1723,6 +1720,12 @@ void GDScriptAnalyzer::resolve_function_signature(GDScriptParser::FunctionNode *
 #endif // TOOLS_ENABLED
 	}
 
+#ifdef DEBUG_ENABLED
+	if (p_function->return_type == nullptr) {
+		parser->push_warning(p_function, GDScriptWarning::UNTYPED_DECLARATION, "Function", function_visible_name);
+	}
+#endif
+
 	if (p_function->get_datatype().is_resolving()) {
 		p_function->set_datatype(prev_datatype);
 	}
```

</details>

Reply:
What is sleep, anyways? 😩